api/server.js

```diff
@@ -0,0 +1,167 @@
+// imports
+const express = require("express");
+const morgan = require("morgan");
+const helmet = require("helmet");
+const cors = require("cors");
+
+// DB import
+const userDB = require("../data/helpers/userDb");
+const postDB = require("../data/helpers/postDb");
+
+// create server instance
+const server = express();
+
+// Custom Middleware
+
+function checkAllCaps(req, res, next){
+    const name = req.body.name;
+    const arr = name.split(" ");
+    let result;
+    for(let i = 0; i < arr.length; i++){
+        if(arr[i][0] !== arr[i][0].toUpperCase()){
+            result = false;
+        }else{
+            result = true;
+        }
+    }
+    if(result){
```

Comment:
I'm not sure if this works as you might expect with the scope of `result` and the fact that you're changing it with each iteration of your for loop above. This will only reflect that final change, nothing before it.